WIP - File naming convention changes

mairh opened the pull request:
- React components are in PascalCase now
- exported functions are in camelCase now
- Graphql file names are now same as the (mutation/query/subscription) function names.

- At the moment, I am getting a bunch of errors when running `yarn watch`. Not sure why it is complaining about `node_modules` folder. @vlasenko @mitjade Could you check on your end?

![screen shot 2017-09-18 at 14 10 51](https://user-images.githubusercontent.com/4072250/30539500-5f6c342c-9c7b-11e7-92ba-085c3f6f97d3.png)

TODO: Refactor the functions in the `jsx`, `js` files to follow strict https://github.com/airbnb/javascript rules.

mairh commented:
Ok. Found 1 prob. OSX doesn't do the lowercase / uppercase casting. I need to rename the file and then change it back to uppercase.

vlasenko commented:
@mairh I see that you have way many files in lowercase instead of PascalCase, despite you reference them by PascalCase

mairh commented:
This is still WIP. It is definitely possible that I have missed something as of now. Could you please point out in the code what are you referring to?

mairh commented:
Yes, I see it now (for example `counter-jsx` vs `Counter.jsx`). I need to go through all the files that I have changed from lower case to PascalCase without the actual file name change

vlasenko commented:
@mairh Yes

mitjade commented:
@mairh Ware you able to figure out what is causing the errors?

mitjade commented:
@mairh I see that the build passes. But I still get errors when I run this branch. Same as the ones you mentioned above.

mairh commented:
yeah, I fixed the build errors related to filenames. However, I am a bit confused about the `node_module` errors I am getting in this branch.

I tried recreating the `yarn.lock` and deleted `node_modules` folder. Also, `package.json` file is exactly the same as in master.

@vlasenko Do you have any idea what could be the reason?

mitjade commented:
@mairh I think I might be on to something. I started renaming file one by one and got this error when I mistyped `CounterView.jsx` and `CounterView.web.jsx`.

mitjade commented:
`import CounterShow from '../components/CounterView';`

mairh commented:
So `import CounterShow from '../components/CounterView';` works but not `import CounterView from '../components/CounterView';` ? Is that what you're trying to say?

mitjade commented:
@mairh Still investigating.

mitjade commented:
@mairh Found something!

mairh commented:
Yeah. Apparently those .web extensions were added by default when you have npm auto import extension installed on visual studio code.

mitjade commented:
@mairh Is it working for you now? I think the .jsx was the problem.

mairh commented:
I didn’t check yet. I just saw the last commit on my phone. I will check it in 15 mins once I get hold on my laptop again.

mairh commented:
@mitjade @vlasenko Code is working fine now. I will do a merge from the master and do the necessary updated in the tools folder.

vlasenko commented:
@mairh Great, please go ahead

mitjade commented:
@mairh Before merging merge with master after `flow` branch is merged, to fix some quote issues.

mitjade commented on src/client/modules/pageNotFound/containers/PageNotFound.jsx:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
-import PageLayout from '../../../app/page_layout';
+import PageLayout from '../../../app/PageLayout';
 
 const pageNotFound = () => (
```
should we change this to `PageNotFound` to be consistent with imports of this component.

mairh replied:
That is the Page wrapper component, not `PageNotFound` module

mitjade replied:
`PageNotFound.jsx` is not a `PageNotFound` module?

mitjade replied:
I am commenting line 7.

mairh replied:
ah I see now. Yes, I will change that. I was paying attention to the line 5

mitjade replied:
Sorry for the confusion.

mitjade commented:
@mairh Fixed all quote problems and template cli to correctly rename files.

mairh commented:
@mitjade @vlasenko How do I solve the pre-commit prettier lint problem, after I merged the latest master into this PR (flow changes)

![screen shot 2017-09-19 at 11 53 41](https://user-images.githubusercontent.com/4072250/30584080-59135680-9d31-11e7-9ebf-51995e2dced4.png)

mitjade commented:
@mairh I do not see latest `master` (with `flow`) being merged into this PR yet?

mairh commented:
@mitjade Yes, It's because the pre-commit is failing due to the errors I reported and thus it doesn't allow you to push any changes.

vlasenko commented:
@mairh @mitjade Let me handle that

vlasenko commented:
@mairh @mitjade Merged master in and added `flow-typed` dir to `.eslintignore'

vlasenko commented:
@mairh @mitjade If you are done, lets merge it into master

mitjade commented:
@vlasenko @mairh I am all for merging this now.

mairh commented:
Let's merge. We can continue further updates in a new PR
